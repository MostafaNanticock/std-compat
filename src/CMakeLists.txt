add_subdirectory(common)

# add_subdirectory(any)
# add_subdirectory(expected) # WIP
# add_subdirectory(filesystem)
# add_subdirectory(optional)
# add_subdirectory(span)
add_subdirectory(variant)
