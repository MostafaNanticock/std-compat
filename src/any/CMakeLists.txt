std_compat_add_module(any
    SOURCES
        ${CMAKE_CURRENT_SOURCE_DIR}/include/any
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/any.hpp

    DEPENDS
        StdCompat::common

    COMPILE_DEFINITIONS
        any_CONFIG_SELECT_ANY=1
)
