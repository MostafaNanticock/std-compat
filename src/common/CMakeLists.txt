std_compat_add_module(common
    SOURCES
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/crt_patches.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_patches/patches.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_patches/patches_mingw.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_patches/mingw_4.8_std_to_string.h

        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/detect_compiler.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/compiler_clang.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/compiler_gcc.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/compiler_msvc.h

        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/detect_crt.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_apple.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_bionic.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_libcxx.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_libstdcxx.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_mingw.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_msvc.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_musl.h
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/crt_newlib.h
)
