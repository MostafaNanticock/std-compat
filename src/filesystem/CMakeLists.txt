std_compat_add_module(filesystem
    SOURCES
        ${CMAKE_CURRENT_SOURCE_DIR}/include/filesystem
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/filesystem.hpp

    DEPENDS
        StdCompat::common

    COMPILE_DEFINITIONS
        filesystem_CONFIG_SELECT_FILESYSTEM=1
)
