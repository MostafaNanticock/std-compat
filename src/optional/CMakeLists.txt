std_compat_add_module(optional
    SOURCES
        ${CMAKE_CURRENT_SOURCE_DIR}/include/optional
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/optional.hpp

    DEPENDS
        StdCompat::common
)
