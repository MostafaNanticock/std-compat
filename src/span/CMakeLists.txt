set(INCLUDE_DIRS
    ${CMAKE_CURRENT_SOURCE_DIR}/include
)

add_library(StdCompat_span INTERFACE)

target_sources(StdCompat_span
INTERFACE
    ${CMAKE_CURRENT_SOURCE_DIR}/include/span
    ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/span.hpp
)

target_link_libraries(StdCompat_span
INTERFACE
    StdCompat_common
)

# Add include directories to consumers, BEFORE system includes
target_include_directories(StdCompat_span
    INTERFACE
        $<BUILD_INTERFACE:${INCLUDE_DIRS}>
)

target_compile_definitions(StdCompat_span
    INTERFACE
        span_CONFIG_SELECT_SPAN=1
)
