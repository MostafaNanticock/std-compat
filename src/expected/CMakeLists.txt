std_compat_add_module(expected
    SOURCES
        ${CMAKE_CURRENT_SOURCE_DIR}/include/expected
        ${CMAKE_CURRENT_SOURCE_DIR}/include/expected_compatibility
        ${CMAKE_CURRENT_SOURCE_DIR}/include/std-compat/internal/expected.hpp

    DEPENDS
        StdCompat::common

    COMPILE_DEFINITIONS
        nsel_CONFIG_SELECT_EXPECTED=1
)
